FLT/for_mathlib/IsCentralSimple.lean

```diff
@@ -30,8 +30,26 @@ structure IsCentralSimple
 
 variable (K : Type u) [Field K]
 
+open Matrix in
 theorem MatrixRing.isCentralSimple (ι : Type v) (hι : Fintype ι) (hnonempty : Nonempty ι) [DecidableEq ι] :
-    IsCentralSimple K (Matrix ι ι K) := sorry
+    IsCentralSimple K (Matrix ι ι K) where
+  is_central d hd := by
+    rw [Subring.mem_center_iff] at hd
+    obtain ⟨k⟩ := hnonempty
+    use d k k
+    ext i j
+    simp_rw [algebraMap_matrix_apply, Algebra.id.map_eq_id, RingHom.id_apply]
+    split_ifs with h
+    · subst h
+      specialize hd (stdBasisMatrix k i (1 : K))
+      rw [← ext_iff] at hd
+      specialize hd k i
+      simpa using hd
+    · specialize hd (stdBasisMatrix j i (1 : K))
+      rw [← ext_iff] at hd
+      specialize hd j j
+      simpa [Ne.symm h] using hd
```

Comment:
```suggestion
  is_central d hd := by
    rw [Subring.mem_center_iff] at hd
    convert mem_range_scalar_of_commute_stdBasisMatrix (M := d) fun i j hij => hd _
    simp_rw [Set.mem_range, eq_comm]
    rfl
```
Change the definition to use `Set.range` instead and you can knock three more lines off the proof.

Reply:
Thanks!